pytorch_lightning/callbacks/finetuning.py

```diff
@@ -82,6 +84,36 @@ def finetune_function(self, pl_module, current_epoch, optimizer, optimizer_idx):
                     )
     """
 
+    def __init__(self):
+        self._internal_state = {}
```

Comment:
Can you define the proper type here? If I'm not mistaken it should be something like

`Dict[int, Dict[str, Tensor]]`